dotCMS/src/main/java/com/dotmarketing/factories/MultiTreeAPIImpl.java

```diff
@@ -317,25 +319,46 @@ private List<MultiTree> getMultiTreeByVariantWithoutFallback(final String parent
     @CloseDBIfOpened
     @Override
     public Set<String> getPersonalizationsForPage(final String pageID) throws DotDataException {
-        IHTMLPage pageId=APILocator.getHTMLPageAssetAPI().fromContentlet(APILocator.getContentletAPI().findContentletByIdentifierAnyLanguage(pageID));
+        IHTMLPage pageId = APILocator.getHTMLPageAssetAPI()
+                .fromContentlet(APILocator.getContentletAPI().findContentletByIdentifierAnyLanguage(pageID));
         return getPersonalizationsForPage(pageId);
     }
     
     @CloseDBIfOpened
     @Override
     public Set<String> getPersonalizationsForPage(final IHTMLPage page) throws DotDataException {
-        final Set<String> personas=new HashSet<>();
+        return getPersonalizationsForPage(page, VariantAPI.DEFAULT_VARIANT.name());
+    }
 
-        final Table<String, String, Set<PersonalizedContentlet>> pageContents = Try.of(()-> getPageMultiTrees(page, false)).getOrElseThrow(e->new DotRuntimeException(e));
+    @CloseDBIfOpened
+    @Override
+    public Set<String> getPersonalizationsForPage(final IHTMLPage page, final String variantName) throws DotDataException{
+        final Set<String> personalizationsForPagVariant = getPersonalizationsForPageInner(page, variantName);
+        Set<String> personalizationsForPagDefault = Collections.emptySet();
```

Comment:
Make it final if it's possible please.

Reply:
done https://github.com/dotCMS/core/pull/26026/commits/4674148f42b0d4e37169b445b3a863c1d41c25a3